helm/kyverno-policies-ux/values.yaml

```diff
@@ -6,17 +6,17 @@ provider:
 
 preventDeletionLabelPolicy:
   resources:
-    - cluster.x-k8s.io/v1beta1/Cluster
-    - cluster.x-k8s.io/v1beta1/MachineDeployment
-    - cluster.x-k8s.io/v1beta1/MachinePool
-    - infrastructure.giantswarm.io/v1alpha3/AWSCluster
-    - infrastructure.giantswarm.io/v1alpha3/AWSControlPlane
-    - infrastructure.giantswarm.io/v1alpha3/AWSMachineDeployment
-    - infrastructure.giantswarm.io/v1alpha3/G8sControlPlane
-    - infrastructure.cluster.x-k8s.io/v1beta1/AzureMachinePool
-    - infrastructure.cluster.x-k8s.io/v1beta1/VSphereCluster
-    - application.giantswarm.io/v1alpha1/App
-    - security.giantswarm.io/v1alpha1/Organization
-    - v1/Namespace
-    - v1/Secret
-    - v1/ConfigMap
+    - Cluster
```

Comment:
This could be many things, for example [Crossplane AWS `Cluster`](https://doc.crds.dev/github.com/crossplane/provider-aws/dax.aws.crossplane.io/Cluster/v1alpha1@v0.42.0). Likewise, Crossplane also shadows built-in names like [`Deployment`](https://doc.crds.dev/github.com/crossplane/provider-aws/apigateway.aws.crossplane.io/Deployment/v1alpha1@v0.42.0). I think we should be specific here and take group+kind, but not the version, since the group rarely ever changes.

Reply:
Yes, that's true. But I don't think that unintentionally targeting different resources is a bad thing. It will not have any effect unless you add the label to the resource.

Originally, we wanted to target all possible resources, but Kyverno (or some Kyverno policy) does not allow to create policies that match all resources, because of performance considerations.

I feel like adding the group would complicate the helm template too much.

Reply:
Fair point. I don't have a strong opinion, as long as performance and memory usage of Kyverno isn't crazy. We can keep it simple.

Reply:
In my tests performance and memory did not seem like a problem.

I would say let's continue like this. If it becomes a problem, we can still change it to group+kind.